### PR TITLE
Polishing

### PR DESCRIPTION
**Pull request by je-ik**

### CODE REVIEW

File: direct/core/src/test/java/cz/o2/proxima/direct/storage/InMemStorage.java [hunk 1]
```diff
@@ -411,16 +411,20 @@ private void handleFlushDataBaseOnPosition(
...
+                if (watermark.getWatermark()
+                    >= (Watermarks.MAX_WATERMARK - BOUNDED_OUT_OF_ORDERNESS)) {
+                  observer.onCompleted();
+                  getObservers(getUri()).remove(consumerId);
+                  onIdleRef.get().cancel(true);
+                  killSwitch.set(true);
+                }
               }
+            } catch (Throwable ex) {
+              log.warn("Exception in onIdle", ex);
```
Comment (dmvk) [inline direct/core/src/test/java/cz/o2/proxima/direct/storage/InMemStorage.java:427]: are you sure about this?

Comment (je-ik) [reply direct/core/src/test/java/cz/o2/proxima/direct/storage/InMemStorage.java:427]: What's wrong with that?

Comment (je-ik) [reply direct/core/src/test/java/cz/o2/proxima/direct/storage/InMemStorage.java:427]: If exception is not caught, the `onIdle` mechanism breaks.

Comment (dmvk) [reply direct/core/src/test/java/cz/o2/proxima/direct/storage/InMemStorage.java:427]: that we are basically ignoring the exception here, should this be handled somehow?

Comment (je-ik) [reply direct/core/src/test/java/cz/o2/proxima/direct/storage/InMemStorage.java:427]: Not sure how. Any suggestions?

**Pull request merged**